src/libexpr/eval-gc.cc

```diff
@@ -143,6 +144,39 @@ class BoehmDisableGC
     };
 };
 
+static size_t getFreeMem()
+{
+    /* On Linux, use the `MemAvailable` or `MemFree` fields from
+       /proc/cpuinfo. */
+#  if __linux__
+    {
+        std::unordered_map<std::string, std::string> fields;
+        for (auto & line : tokenizeString<std::vector<std::string>>(readFile("/proc/meminfo"), "\n")) {
+            auto colon = line.find(':');
+            if (colon == line.npos)
+                continue;
+            fields.emplace(line.substr(0, colon), trim(line.substr(colon + 1)));
+        }
+
+        auto i = fields.find("MemAvailable");
+        if (i == fields.end())
+            i = fields.find("MemFree");
+        if (i != fields.end()) {
```

Comment:
I think this number might be misleading on zfs (which is quite popular with NixOS user) as zfs ARC allocations are not accounted in page cache. There was a trick though that htop uses to get a more correct number.

Reply:
```
# Get total memory in kB
total_mem=$(grep MemTotal /proc/meminfo | awk '{print $2}')

# Get free memory in kB
free_mem=$(grep MemFree /proc/meminfo | awk '{print $2}')

# Get available memory in kB
avail_mem=$(grep MemAvailable /proc/meminfo | awk '{print $2}')

# Get ARC memory in bytes from arcstats
arc_mem=$(awk '/^size/ {print $3}' /proc/spl/kstat/zfs/arcstats)

# Convert total memory to bytes for consistency
total_mem=$((total_mem * 1024))

# Calculate used memory excluding ARC (in bytes)
used_mem_excluding_arc=$((total_mem - free_mem * 1024 - avail_mem * 1024 + arc_mem))

# Convert result to a more readable format (kB)
used_mem_excluding_arc_kb=$((used_mem_excluding_arc / 1024))

# Print the result
echo "Used memory excluding ARC: $used_mem_excluding_arc_kb kB"
```

Reply:
Here is an example arcstats: https://gist.github.com/Mic92/b2d37f5f8ed7f93dff432030ec8f1c67